src/blueprint/from_config.rs

```diff
@@ -20,11 +21,22 @@ use crate::json::JsonSchema;
 use crate::lambda::Expression::Literal;
 use crate::lambda::Lambda;
 use crate::request_template::RequestTemplate;
-use crate::valid::{OptionExtension, Valid as ValidDefault, ValidExtensions, ValidationError, VectorExtension};
+use crate::valid::{Cause, OptionExtension, Valid as ValidDefault, ValidExtensions, ValidationError, VectorExtension};
 use crate::{blueprint, config};
 
 type Valid<A> = ValidDefault<A, String>;
 
+impl<A> From<ValidationError<&str>> for Valid<A> {
+  fn from(value: ValidationError<&str>) -> Self {
+    let message = value
+      .as_vec()
+      .iter()
+      .map(|c| Cause::new(c.message.to_owned()))
+      .collect();
+    Valid::fail_cause(message)
```

Comment:
@amitksingh1490 this impl seem fishy. Let me know if you have any suggestions to improve it.

Reply:
The main issue with it is that creating a cause here ignores description and trace. 

I'm thikning something like following,
```diff
 use crate::lambda::Expression::Literal;
 use crate::lambda::Lambda;
 use crate::request_template::RequestTemplate;
-use crate::valid::{Cause, OptionExtension, Valid as ValidDefault, ValidExtensions, ValidationError, VectorExtension};
+use crate::valid::{OptionExtension, Valid as ValidDefault, ValidExtensions, ValidationError, VectorExtension};
 use crate::{blueprint, config};
 
 type Valid<A> = ValidDefault<A, String>;
 
 impl<A> From<ValidationError<&str>> for Valid<A> {
   fn from(value: ValidationError<&str>) -> Self {
-    let message = value
-      .as_vec()
-      .iter()
-      .map(|c| Cause::new(c.message.to_owned()))
-      .collect();
+    let message = value.as_vec().iter().map(|c| c.clone().into()).collect();
     Valid::fail_cause(message)
   }
 }
diff --git a/src/valid/cause.rs b/src/valid/cause.rs
index bf5b200..fb8a6c3 100644
--- a/src/valid/cause.rs
+++ b/src/valid/cause.rs
@@ -29,6 +29,16 @@ impl<E: Display> Display for Cause<E> {
   }
 }
 
+impl From<Cause<&str>> for Cause<String> {
+  fn from(value: Cause<&str>) -> Self {
+    Cause {
+      message: value.message.to_owned(),
+      description: value.description.map(|d| d.to_owned()),
+      trace: value.trace,
+    }
+  }
+}
+
 impl<E> Cause<E> {
   pub fn new(e: E) -> Self {
     Cause { message: e, description: None, trace: VecDeque::new() }
```

How about it? :thinking:

Reply:
Ya this seems ok, Please go ahead and add a test for this change

Reply:
Thanks, @amitksingh1490. Sorry but the one suggested in the comment or the existing code? :slightly_smiling_face: 

Also, test for the change is already added.

Reply:
I think you meant the one suggested in the comment.